src/services/quiz.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Question } from '../database/entities/Question';
+import { QuestionOption } from '../database/entities/QuestionOption';
+import type { UserPayload } from '../typings/auth';
+import { Errors } from '../utils/error.util';
+import type {
+    questionOptionType, quizType
+} from '../validations/quiz.validate';
+import { courseService } from './course.service';
+import { moduleService } from './module.service';
+
+class QuizService {
+
+    async addNewQuestion(
+        courseId: number,
+        quizId: number,
+        userId: UserPayload['userId'],
+        rawQuestion: quizType) {
+
+        const course = await courseService.get(courseId);
+
+        if (course.instructorId !== userId) {
+            throw Errors.NO_PERMISSION;
+        }
+
+        await moduleService.getQuiz(quizId);
+
+        const questionData = {
+            quizId: quizId,
+            question: rawQuestion.question,
+        };
+
+        const answersData: questionOptionType[] =
```

Comment:
ini tuh list of question option ya? knp ga questionOptions aja nama variabelnya, biar ga ambigu wkwk